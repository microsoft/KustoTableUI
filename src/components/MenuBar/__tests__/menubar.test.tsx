import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { MenuBar } from '..';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';

describe('MenuBar Snapshot UI Tests', () => {
  const history = createBrowserHistory();
  let component: ReactTestRenderer;

  afterEach(() => component.unmount());

  test.each`
    title                  | userName       | scenarioName
    ${null}                | ${null}        | ${'Null Title and User Name'}
    ${undefined}           | ${undefined}   | ${'Undefined Title and User Name'}
    ${''}                  | ${''}          | ${'Empty String Title and User Name'}
    ${'Title for the App'} | ${'Test User'} | ${'Normal Title and User Name'}
  `('Test MenuBar Rendering $scenarioName', ({ title, userName }) => {
    component = renderer.create(
      <Router history={history}>
        <MenuBar title={title} userName={userName} />
      </Router>,
    );
    const menuBar = component.toJSON();
    expect(menuBar).toMatchSnapshot();
  });
});
