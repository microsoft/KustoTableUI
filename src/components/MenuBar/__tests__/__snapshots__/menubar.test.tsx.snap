// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBar Snapshot UI Tests Test MenuBar Rendering Empty String Title and User Name 1`] = `
<div
  className="menubar-container"
>
  <div
    className="menubar-container-left"
  >
    <a
      className="menubar-title"
      href="/home"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="menubar-icon"
        src="logo.svg"
      />
    </a>
    <a
      className="menubar-item-link"
      href="/Example1"
      onClick={[Function]}
    >
      Example1
    </a>
    <a
      className="menubar-item-link"
      href="/Example2"
      onClick={[Function]}
    >
      Example2
    </a>
  </div>
  <div
    className="menubar-container-right"
  >
    <span
      className="menubar-username"
    />
  </div>
</div>
`;

exports[`MenuBar Snapshot UI Tests Test MenuBar Rendering Normal Title and User Name 1`] = `
<div
  className="menubar-container"
>
  <div
    className="menubar-container-left"
  >
    <a
      className="menubar-title"
      href="/home"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="menubar-icon"
        src="logo.svg"
      />
      Title for the App
    </a>
    <a
      className="menubar-item-link"
      href="/Example1"
      onClick={[Function]}
    >
      Example1
    </a>
    <a
      className="menubar-item-link"
      href="/Example2"
      onClick={[Function]}
    >
      Example2
    </a>
  </div>
  <div
    className="menubar-container-right"
  >
    <span
      className="menubar-username"
    >
      Test User
    </span>
  </div>
</div>
`;

exports[`MenuBar Snapshot UI Tests Test MenuBar Rendering Null Title and User Name 1`] = `
<div
  className="menubar-container"
>
  <div
    className="menubar-container-left"
  >
    <a
      className="menubar-title"
      href="/home"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="menubar-icon"
        src="logo.svg"
      />
    </a>
    <a
      className="menubar-item-link"
      href="/Example1"
      onClick={[Function]}
    >
      Example1
    </a>
    <a
      className="menubar-item-link"
      href="/Example2"
      onClick={[Function]}
    >
      Example2
    </a>
  </div>
  <div
    className="menubar-container-right"
  >
    <span
      className="menubar-username"
    />
  </div>
</div>
`;

exports[`MenuBar Snapshot UI Tests Test MenuBar Rendering Undefined Title and User Name 1`] = `
<div
  className="menubar-container"
>
  <div
    className="menubar-container-left"
  >
    <a
      className="menubar-title"
      href="/home"
      onClick={[Function]}
    >
      <img
        alt="logo"
        className="menubar-icon"
        src="logo.svg"
      />
    </a>
    <a
      className="menubar-item-link"
      href="/Example1"
      onClick={[Function]}
    >
      Example1
    </a>
    <a
      className="menubar-item-link"
      href="/Example2"
      onClick={[Function]}
    >
      Example2
    </a>
  </div>
  <div
    className="menubar-container-right"
  >
    <span
      className="menubar-username"
    />
  </div>
</div>
`;
